docs/advanced_topics/performance.md

```diff
@@ -44,6 +44,14 @@ CACHES = {
 }
 ```
 
+### Page URLs
+
+To fully resolve the URL of a page, Wagtail uses information from lots of different sources, namely the current site and request.
+
+The methods used to get the URL of a `Page` such as `Page.get_url` and `Page.get_full_url` optionally accept extra arguments for `request` and `current_site`. This enables caching of underlying site-level URL information throughout the request cycle, thereby significantly reducing the number of cache or SQL queries your site will generate for a given page load.
```

Comment:
I thought mentioning navigation menus etc might help highlight the impact this can have, but again, that required a bit of breaking down to keep things readable. Hope this is okay:
```suggestion
The methods used to get the URL of a `Page` such as `Page.get_url` and `Page.get_full_url` accept optional `request` and `current_site` arguments which, when provided, allow a lot of underlying site-level URL data to be reused for the current request. When taking into account navigation menu generation, plus any links that appear in page content, providing `request` or `current_site` can result in a drastic overall reduction in the number of cache/database queries that are required to render a page.
```